Assets/Project/Scripts/MenuSelectScene/Record/GeneralRecordDirector.cs

```diff
@@ -0,0 +1,297 @@
+                }
+            }
+
+            _shareButton.onClick.AddListener(ShareGeneralRecord);
+            _individualButton.onClick.AddListener(_recordDirector.MoveToRight);
+            _clearStageNum.text = GetClearStageNum();
+            _stageNum.text = GetStageNum();
+            _clearStageGauge.fillAmount = GetClearStagePercentage();
+            _clearStageIndicator.GetComponent<RectTransform>().localPosition = GetClearStageIndicatorPosition();
+            _playNum.text = GetPlayNum();
+            _playDays.text = GetPlayDays();
+            _flickNum.text = GetFlickNum();
+            _failureNum.text = GetFailureNum();
+
+            SetupFailureReasonGraph();
+        }
+
+        private static void ShareGeneralRecord()
+        {
+            Application.OpenURL("https://twitter.com/intent/tweet?hashtags=Treevel");
+        }
+
+        private string GetClearStageNum()
+        {
+            var clearStageNum = _stageStatuses.Select(stageStatuses => stageStatuses.successNum > 0 ? 1 : 0).Sum();
+
+            return clearStageNum.ToString();
+        }
+
+        private string GetStageNum()
+        {
+            var stageNum = _stageStatuses.Count;
+
+            return stageNum.ToString();
+        }
+
+        private float GetClearStagePercentage()
+        {
+            var stageNum = _stageStatuses.Count;
+            var clearStageNum = _stageStatuses.Select(stageStatuses => stageStatuses.successNum > 0 ? 1 : 0).Sum();
+
+            var percentage = (float) clearStageNum / stageNum;
+
+            return percentage;
+        }
+
+        private Vector3 GetClearStageIndicatorPosition()
+        {
+            var radius = _clearStageGauge.GetComponent<RectTransform>().rect.width / 2;
+            var angle = GetClearStagePercentage() * 2 * Mathf.PI;
+
+            // FIXME: 0.95 はゲージの太さを考慮するための Magic Number です
+            var x = radius * Mathf.Sin(angle) * 0.95f;
+            var y = radius * Mathf.Cos(angle) * 0.95f;
+
+            return new Vector3(x, y);
+        }
+
+        private string GetPlayNum()
+        {
+            var playNum = _stageStatuses.Select(stageStatuses => stageStatuses.challengeNum).Sum();
+
+            return playNum.ToString();
+        }
+
+        private static string GetPlayDays()
+        {
+            var playDays = RecordData.Instance.StartupDays;
+
+            return playDays.ToString();
+        }
+
+        private string GetFlickNum()
+        {
+            var flickNum = _stageStatuses.Select(stageStatuses => stageStatuses.flickNum).Sum();
+
+            return flickNum.ToString();
+        }
+
+        private string GetFailureNum()
+        {
+            var failureNum = _stageStatuses.Select(stageStatuses => stageStatuses.failureNum).Sum();
+
+            return failureNum.ToString();
+        }
+
+        private void SetupFailureReasonGraph()
+        {
+            // 失敗回数の合計
+            float sum = RecordData.Instance.FailureReasonCount
```

Comment:
```suggestion
            float sum = RecordData.Instance.FailureReasonCount.Sum(e => e.Value);
```
LinQで手に入るの(Key, Value)のペアらしいからdic(tionary)と名付けるのは違和感。
elementとか

Reply:
この書き方できたのか．学び！
- 6378bad